gdeep/models/utils.py

```diff
@@ -70,6 +70,6 @@ def __call__(
                 the output tensor of the module
         """
         if isinstance(module_out, tuple):
-            self.outputs.append(module_out[0].detach())
+            self.outputs.append(module_out[0])#.detach())
```

Comment:
Thanks for the comment, this is a very good point indeed. I do understand the consequences and I agree what was proposed is not a good solution. To be honest, I forgot about this myself. It was just a quick and dirty fix to make the BERT models work and since it worked and seemingly didn't break any tests I just proceeded without worrying about it.

A good solution would keep the old functionality where applicable, in the same spirit as the change in trainer.py. I believe this has now been fixed.